controllers/ingestion/http/http.go

```diff
@@ -0,0 +1,79 @@
+package http
+
+import (
+	"bytes"
+	"io/ioutil"
+	"net/http"
+)
+
+// DruidHTTP interface
+type DruidHTTP interface {
+	Do() (*Response, error)
+}
+
+// HTTP client
+type Client struct {
+	Method     string
+	URL        string
+	HTTPClient http.Client
+	Body       []byte
+	Auth       Auth
+}
+
+func NewHTTPClient(method, url string, client http.Client, body []byte, auth Auth) DruidHTTP {
```

Comment:
Suggest using the `https://github.com/go-playground/validator` package to validate the object inside the "New" functions